lib/fabrik/DirectorManager.js

```diff
@@ -237,6 +270,7 @@ class DirectorManager extends BaseManager {
             });
         }
       })
+      .then(() => this.executeActions(CONST.SERVICE_LIFE_CYCLE.PRE_DELETE, deploymentName))
```

Comment:
It could be a good idea to invoke predelete hook before invoking the deprovision hook on the agent, as later down the line if required we could then pass predelete hook response to deprovision hook and agent can make use of that info someway.

Reply:
Done!